### PR TITLE
interactive: added label to display selected queries to user (5/7)

### PR DESCRIPTION
**Pull request by dshaaban01**
added label to display selected queries to user.

image

### CODE REVIEW

File: tests/interactive/test_app.py [hunk 1]
```diff
@@ -9,10 +13,12 @@
 
 
 @pytest.mark.asyncio()
-async def test_input():
+async def tests_input():
     """Test pressing keys has the desired result."""
-    app = InputApp()
-    async with app.run_test() as pilot:  # pyright: ignore
+    async with InputApp().run_test() as pilot:
+        pilot: Pilot[None] = pilot
```
Comment (superlopuh) [inline tests/interactive/test_app.py:19]: don't need this any more

File: xdsl/interactive/app.py [hunk 2]
```diff
@@ -52,45 +54,85 @@ class InputApp(App[None]):
         },
     )
 
-    current_module: reactive[ModuleOp | Exception | None] = reactive(None)
+    current_module = reactive[ModuleOp | Exception | None](None)
```
Comment (superlopuh) [inline xdsl/interactive/app.py:57]: isn't this in a different PR?

File: xdsl/interactive/app.tcss [hunk 3]
```diff
@@ -9,6 +9,26 @@
     align: center middle;
 }
 
+# ScrollableContainer(Label)
+#selected_passes {
+    margin: 1 1 0 1;
+    border: heavy $warning-lighten-1;
+    border-title-color: $primary;
+    # grid-columns: 75% 25%;
```
Comment (superlopuh) [inline xdsl/interactive/app.tcss:17]: please remove commented line

**Review by superlopuh — CHANGES_REQUESTED** [review]
It feels a little off for the borders to be unaligned above and below, could you please make them line up?

**Comment by webmiche** [comment]
I keep getting confused with all of these PRs and how they are ordered. Could you add an index to the PR title? Something like `interactive: (3/8) ...` 8or `interactive: (3/?) ...` if you don't know how many there are.

**Comment by webmiche** [comment]
Classic, supervisors giving conflicting advice :wink:

**Comment by superlopuh** [comment]
Please merge main and I'll take a look

File: xdsl/interactive/app.py [hunk 4]
```diff
@@ -76,6 +79,13 @@ def __init__(self):
         self.passes_selection_list = SelectionList(id="passes_selection_list")
         super().__init__()
 
+    query_label = Label("", id="selected_passes_label")
+    """Display's user selected passes"""
```
Comment (superlopuh) [inline xdsl/interactive/app.py:83]:
```suggestion
    """Display's user selected passes."""
```

File: xdsl/interactive/app.py [hunk 5]
```diff
@@ -76,6 +79,13 @@ def __init__(self):
...
+
+    passes_selection_list: SelectionList[type[ModulePass]] = SelectionList(
```
Comment (superlopuh) [inline xdsl/interactive/app.py:85]: please move to init

Comment (superlopuh) [reply xdsl/interactive/app.py:85]: also please move variable declarations above init

**Review by superlopuh — CHANGES_REQUESTED** [review]
looking good, just want to avoid the same class variable situation with the new setter

File: xdsl/interactive/app.py [hunk 6]
```diff
@@ -67,13 +68,18 @@ class InputApp(App[None]):
...
     passes_selection_list: SelectionList[type[ModulePass]]
 
+    selected_query_label = Label("", id="selected_passes_label")
+    """Display selected passes"""
+
     def __init__(self):
         self.input_text_area = TextArea(id="input")
         self.output_text_area = OutputTextArea(id="output")
         self.passes_selection_list = SelectionList(id="passes_selection_list")
+        self.passes_selection_list: SelectionList[type[ModulePass]] = SelectionList(
```
Comment (superlopuh) [inline xdsl/interactive/app.py:80]:
```suggestion
        self.selected_query_label = Label("", id="selected_passes_label")
```

File: xdsl/interactive/app.py [hunk 7]
```diff
@@ -67,13 +68,18 @@ class InputApp(App[None]):
 
     input_text_area: TextArea
     output_text_area: OutputTextArea
-
     passes_selection_list: SelectionList[type[ModulePass]]
 
+    selected_query_label = Label("", id="selected_passes_label")
```
Comment (superlopuh) [inline xdsl/interactive/app.py:73]:
```suggestion
    selected_query_label: Label
```

**Review by superlopuh — CHANGES_REQUESTED** [review]

**Review by superlopuh — APPROVED** [review]

**Pull request merged**